### PR TITLE
Include raw jwt in errors

### PR DESCRIPTION
**Pull request by ottokruse**
*Issue #, if available:* #24

*Description of changes:* Ability to include the raw JWT in thrown errors. Also, instead of a generic `AssertionError` more concrete errors are now thrown (e.g. `CognitoJwtInvalidGroupError`) to make it easier to recognize the concrete error at hand.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,6 +142,7 @@ Supported parameters are:
 - `scope` (optional): verify that the JWT's `scope` claim matches your expectation (only of use for access tokens). Provide a string, or an array of strings to allow multiple scopes (i.e. one of these scopes must match the JWT). See also [Checking scope](#Checking-scope).
 - `graceSeconds` (optional, default `0`): to account for clock differences between systems, provide the number of seconds beyond JWT expiry (`exp` claim) or before "not before" (`nbf` claim) you will allow.
 - `customJwtCheck` (optional): your custom function with additional JWT (and JWK) checks to execute (see also below).
+- `includeRawJwtInErrors` (optional): set to `true` if you want to peek inside the invalid JWT when verification fails. Refer to: [Peek inside invalid JWTs](#peek-inside-invalid-jwts).
```
Comment (elorzafe) [inline README.md:145]: Maybe is worth to mention default is `false`

File: src/cognito-verifier.ts [hunk 2]
```diff
@@ -291,14 +314,24 @@ export class CognitoJwtVerifier<
...
-    const payload = super.verifySync(...args);
-    const issuerConfig = this.getIssuerConfig(payload.iss);
-    const verifyProperties = {
-      ...issuerConfig,
-      ...args[1],
-    };
-    validateCognitoJwtFields(payload, verifyProperties);
-    return payload as CognitoIdOrAccessTokenPayload<IssuerConfig, T>;
+    const { decomposedJwt, jwksUri, verifyProperties } =
+      this.getVerifyParameters(args[0], args[1]);
```
Comment (elorzafe) [inline src/cognito-verifier.ts:318]: How`args` parameters are organized?

File: src/error.ts [hunk 3]
```diff
@@ -1,8 +1,51 @@
...
+ */
 export abstract class JwtBaseError extends Error {}
 
+/**
+ * Interface for an error that is raised because an actual value does not match with the expected value
+ */
+interface AssertionError extends JwtBaseError {
+  failedAssertion: {
+    actual: unknown;
+    expected?: string | string[];
```
Comment (elorzafe) [inline src/error.ts:17]: Can this be an JSON? What is the assumption here, maybe is worth to mention.

File: tests/unit/assert.test.ts [hunk 4]
```diff
@@ -1,43 +1,88 @@
...
   test("assert equals requires non-empty value", () => {
-    const statement = () => assertStringEquals("test", "", "two");
-    expect(statement).toThrow(AssertionError);
-    expect(statement).toThrow("Missing test. Expected: two");
+    const actual = "";
+    const expected = "two";
+    const statement = () => assertStringEquals("Test", actual, expected);
+    expect.assertions(3);
+    expect(statement).toThrow(FailedAssertionError);
+    expect(statement).toThrow("Missing Test. Expected: two");
```
Comment (elorzafe) [inline tests/unit/assert.test.ts:17]: I didnt know this was possible, cool!

**Review by elorzafe — COMMENTED** [review]
Minor suggestions, but looks very good. We can chat more offline

File: README.md (hunk 1 above)
Comment (ottokruse) [reply README.md]: Done

File: src/error.ts (hunk 3 above)
Comment (ottokruse) [reply src/error.ts:17]: Sorry mate what do you mean exactly?

File: src/cognito-verifier.ts (hunk 2 above)
Comment (ottokruse) [reply src/cognito-verifier.ts]: Destructured them now for better readability. Does that solve your concern?

File: tests/unit/assert.test.ts (hunk 4 above)
Comment (ottokruse) [reply tests/unit/assert.test.ts:17]: 😎

**Review by leelalagudu — APPROVED** [review]

**Review by elorzafe — APPROVED** [review]
LGTM thanks @ottokruse

**Pull request merged**